### DISCUSSION TITLE
Can't compile this Hello World program. Make not found.

### DISCUSSION THREAD

**Discussion by NealIRC** [eclipse-cdt/cdt#611] [Q&A]
So I downloaded eclipse, and put in this simple "Hello World" code:

#include <iostream>

int main() {
    std::cout << "Hello, world!" << std::endl;
    return 0;
}
And this is my build error message:

Error: Program "make" not found in PATH PATH=[C:\TDM-GCC-64\bin]

Note that this is like the 5th thing I downloaded, and each time I updated the Environment Path and such, to the bin folder, where the gcc.exe is and make.exe, but it still can't find "make." My current folder has mingw32.exe.

I already downloaded MinGW, W64DevKit, llvm-minwg, etc. So I'll take suggestions on what to download...

**Comment by jld01** [comment]
@NealIRC did you observe and select the _MinGW GCC_ toolchain when you created the project?

![image](https://github.com/eclipse-cdt/cdt/assets/45816183/a0b1c274-202b-4253-a6e0-1fa2eda137e2)

I suspect the underlying issue here is that CDT did not detect your TDM-GCC installation. CDT attempts to detect various installations of MinGW tooling but does not look for TDM-GCC. You may need to inform CDT of your TDM-GCC installation by setting the `MINGW_HOME` environment variable:

![image](https://github.com/eclipse-cdt/cdt/assets/45816183/7831cdbe-05bd-44e2-99ba-217076a04923)

If you select the _MinGW GCC_ toolchain in the project wizard, the project will use the CDT internal builder by default and you will not need GNU make.

**Comment by NealIRC** [comment]
Okay thanks I got it working, but I have another question. I'll post that as a reply.

Here's how I got it working.

Under right click the project > Properties > C/C++ Build > Tool Chain Editor, under Current toolchain, it was not set to MinGW GCC. It was set to Cygwin GCC, 1 of my earlier paths I set to.

Somewhere under Properties > C/C++ Build > Settings, I also had to add in a "-mwindows" somewhere.

Somehow, initially I was able to Build but not Run, not sure what tweaking I did to make it build and run.

**Comment by NealIRC** [comment]
C++ by default doesn't have Java's equivalent of JOptionPane with the input, so you have to downloading something else like Qt Creator or wxWidgets.

Downloaded Qt Creatpr. Added a new environmental system variable path C:\Qt\6.6.0\mingw_64\include Added the flags -lQt6Core -lQtWidgets -lQtGui -lQtQml -lQtNetwork

Previously tried

#include <QCoreApplication>
#include <QInputDialog>
#include <QMessageBox>
#include <QFile>
#include <QTextStream>

and got the build error

g++ "-IC:\Qt\6.6.0\mingw_64\include" -O0 -g3 -Wall -c -fmessage-length=0 -o WritingApp.o "..\WritingApp.cpp" ..\WritingApp.cpp:1:10: fatal error: QCoreApplication: No such file or directory 1 | #include <QCoreApplication> | ~~~~~~~~~~~~~~~~~

then tried

#include <QtCore/QCoreApplication>
#include <QtWidgets/QInputDialog>
#include <QtWidgets/QMessageBox>
#include <QtCore/QFile>
#include <QtCore/QTextStream>
and got the error

g++ "-IC:\Qt\6.6.0\mingw_64\include" -O0 -g3 -Wall -c -fmessage-length=0 -o WritingApp.o "..\WritingApp.cpp" g++ -L-LC:/Qt/6.6.0/mingw_64/lib -mwindows -o WritingApp.exe WritingApp.o -lg++ -o WritingApp.exe WritingApp.o C:/Qt/6.6.0/mingw_64/lib QtWidgets QtGui QtQml QtNetwork QtCore C:\Users\lonel\Downloads\w64devkit-1.20.0\w64devkit\bin/ld.exe: cannot find C:/Qt/6.6.0/mingw_64/lib: Permission denied C:\Users\lonel\Downloads\w64devkit-1.20.0\w64devkit\bin/ld.exe: cannot find QtWidgets: No such file or directory


When I right click my project, select Properties > C/C++ Build > Settings > Tool Settings > MinGW C++ Linker > what should I have in the -l and -L fields?

For -l, I have

g++ -o WritingApp.exe WritingApp.o C:/Qt/6.6.0/mingw_64/lib QtWidgets QtGui QtQml QtNetwork QtCore

And for +L, I have

-LC:/Qt/6.6.0/mingw_64/lib

**Reply by jld01** [reply]
As a general rule, please create a _new_ GitHub discussion for each topic.

Assuming the library you are using is already built for MinGW, you would typically have a set of library archive files with names such as `libABC.a` and `libDEF.a` in a folder. Within the _MinGW C++ Linker_ settings, you would list the libraries (`-l`) as `ABC` and `DEF` (omitting the `lib` prefix and the `.a` suffix). You would list the folder(s) containing these libraries as the library search path (`-L`).
